src/skeleplex/graph/modify_graph.py

```diff
@@ -0,0 +1,360 @@
+
+
+        if key == EDGE_SPLINE_KEY:
+
+            points1 = edge_attributes1[EDGE_COORDINATES_KEY]
+            points2 = edge_attributes2[EDGE_COORDINATES_KEY]
+
+            #start and end node coordinates
+            #this checking is probably not necessary as we use directed graphs
+            #but just to be sure
+            if (np.allclose(points1[0], start_node) & 
+                np.allclose(points2[0], middle_node)):
+                logger.info('no flip')
+                spline_points = np.vstack((points1, points2))
+                
+            elif (np.allclose(points1[-1], start_node) &
+                   np.allclose(points2[0], middle_node)):
+                logger.info('flip 1')
+                spline_points = np.vstack((np.flip(points1, axis = 0), points2))
+            elif (np.allclose(points1[0], start_node) & 
+                  np.allclose(points2[-1], middle_node)):
+                logger.info('flip 2')
+                spline_points = np.vstack((points1, np.flip(points2, axis = 0)))
+            elif (np.allclose(points1[-1], start_node) & 
+                  np.allclose(points2[-1], middle_node)):
+                logger.info('flip both')
+                spline_points = np.vstack((np.flip(points1, axis = 0),
+                                            np.flip(points2, axis = 0)))    
+            else:
+                warnings.warn('Warning: Edge splines not connected.')
+                spline_points = np.vstack((points1, points2))
+            #sanity check
+            if np.allclose(spline_points[-1], end_node):
+                logger.info('sanity check passed')    
+
+                
+            _, idx = np.unique(spline_points, axis=0, return_index=True)
+            spline_points = spline_points[np.sort(idx)]
+            spline = B3Spline.from_points(spline_points)
+            merge_attributes[key] = spline
+            merge_attributes[EDGE_COORDINATES_KEY] = spline_points
+        if key == START_NODE_KEY:
+            merge_attributes[key] = n1
+        if key == END_NODE_KEY:
+            merge_attributes[key] = n2
+        if key == GENERATION_KEY:
+            merge_attributes[key] = edge_attributes1[key]
+        
+        if key == LENGTH_KEY:
+            merge_attributes[key] = merge_attributes[EDGE_SPLINE_KEY].arc_length
+
+        if key not in  [VALIDATED_KEY, 
+                            EDGE_COORDINATES_KEY, 
+                            EDGE_SPLINE_KEY, 
+                            START_NODE_KEY, 
+                            END_NODE_KEY, 
+                            GENERATION_KEY,
+                            LENGTH_KEY]:
+            logger.warning(('Warning: Attribute {} not merged. '.format(key) ,
+                            'Consider recomputing.'))
+            
+    graph.add_edge(*merge_edge, **merge_attributes)
+    return graph
+
+def delete_edge(SkeletonGraph_obj:SkeletonGraph, edge: Tuple[int, int]):
+        """delete edge."""
```

Comment:
Please add a statement on the nodes are merged.

Reply:
Added comment and visualisation of behaviour